src/c++/simulator/particle.hpp

```diff
@@ -0,0 +1,65 @@
+﻿#pragma once
+
+#include <iostream>
+#include <simulator/geometry.hpp>
+#include <simulator/mechanics.hpp>
+#include <simulator/my_types.hpp>
+#include <simulator/materials.hpp>
+
+struct IParticle
+{
+public:	
+	virtual Entity get_type() = 0;
+	virtual double get_radius() = 0;
+	
+	template<class T*>
+	void set_material(T* material)
```

Comment:
I believe this is not necessary due to inheritance.

Reply:
I think I did like that because I got compilation errors.
I think the template parameter is redundant and there is no need to explicitly pass the template parameter to the method.